src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs

```diff
@@ -115,5 +122,79 @@ private static void EnsurePayloadValueConverter(this IEdmModel model)
                 model.SetPayloadValueConverter(RestierPayloadValueConverter.Default);
             }
         }
+
+        private class RestierRouteConstraint : IHttpRouteConstraint
+        {
+            public ODataPathRouteConstraint Inner { get; private set; }
+
+            public ODataPathRouteConstraint Placeholder { get; private set; }
+
+            public Func<IApi> ApiFactory { get; private set; }
+
+            public IEdmModel FinalModel { get; private set; }
+
+            public RestierRouteConstraint(IODataPathHandler pathHandler, string routeName, Func<IApi> apiFactory)
+            {
+                var conventions = ODataRoutingConventions.CreateDefault();
+                Placeholder = new ODataPathRouteConstraint(pathHandler, new EdmModel(), routeName, conventions);
+                ApiFactory = apiFactory;
+                Task.Delay(5000).ContinueWith(_ => TryInitModel());
```

Comment:
Sorry the Delay is for testing, to make the pipeline truly async. It's not required.